extensions/twenty/src/create-object-record.tsx

```diff
@@ -0,0 +1,76 @@
+import { Action, ActionPanel, Icon, List, useNavigation } from "@raycast/api";
+import { randomUUID } from "crypto";
+
+import twenty from "./services/TwentySDK";
+import { ObjectIcons } from "./enum/icons";
+import { OpenCreateObjectRecordForm } from "./pages";
+import { usePromise } from "@raycast/utils";
+
+export default function CreateObjectRecord() {
+  const { isLoading, data: activeDataModels } = usePromise(async () => {
+    const activeDataModels = await twenty.getActiveDataModels();
+    return activeDataModels;
+  });
+
+  const { push } = useNavigation();
+
+  const standardActiveModel = activeDataModels?.filter((model) => !model.isCustom);
+  const customActiveModel = activeDataModels?.filter((model) => model.isCustom);
+
+  return (
+    <List isLoading={isLoading} navigationTitle="Create Object Record" searchBarPlaceholder="Search Object Record">
+      <List.Section title="Standard Objects">
+        {standardActiveModel?.map((model) => {
+          const { id, description, labelPlural, icon } = model;
+          return (
+            <List.Item
+              id={id}
+              title={labelPlural}
+              subtitle={description ?? ""}
+              actions={
+                <ActionPanel>
+                  <Action
+                    title="Create Record"
+                    icon={Icon.List}
+                    onAction={async () => {
+                      const objectRecordMetadata = await twenty.getRecordFieldsForDataModel(id);
+                      push(OpenCreateObjectRecordForm({ objectRecordMetadata }));
```

Comment:
Because of the `await` there's a small wait until the next view is `push`ed. Since it's not evident item has been pressed, it's possible to fire the action multiple times which means multiple views get pushed.

My suggestions:
1. hide the `Action` or *return* if running AND
2. show an animated `Toast` and/or `isLoading` indicator so user knows they're good to go

Reply:
@xmok I've made changes in getRecordFieldsForDataModel to use FindOne endpoint instead of FIndMany (earlier it was using FindMany) and now next view  is pushed immediately and there is very small time between two views and when I'm having loading indicator,  user won't be able to see cause its too quick. 

should we have toast in this case too ? while to note, I have handle to show Toast when request fails.

Reply:
Unfortunately, for me it's still not immediate.
You don't need to show an indicator if you don't want to but I'd suggest at least returning or hiding the `Action` 
as double view push is still possible. I can see myself easily double-pressing when I'm using Raycast. Notice especially "People" is slow. This is NOT a deal breaker though so we can disregard.

video:

https://github.com/user-attachments/assets/44096837-9cc8-4178-b182-81640efa3fd7